presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/ExtendedHiveMetastore.java

```diff
@@ -89,7 +89,7 @@ default Optional<Table> getTable(MetastoreContext metastoreContext, HiveTableHan
 
     Optional<Partition> getPartition(MetastoreContext metastoreContext, Table table, List<String> partitionValues);
 
-    Optional<List<String>> getPartitionNames(MetastoreContext metastoreContext, String databaseName, String tableName);
+    Optional<List<String>> getPartitionNames(MetastoreContext metastoreContext, Optional<Table> table);
```

Comment:
Why is it Optional ? Why do you want to call getPartitionNames if its empty ?

Reply:
There is an existing check at both places in SemiTransactionalHiveMetastore to see if tableOptional is empty or not. So this method is only called when table object is not empty.